node_modules/oae-core/documentpreview/js/documentpreview.js

```diff
@@ -84,86 +171,85 @@ define(['jquery', 'underscore', 'oae.core', 'lazyload'], function($, _, oae) {
             // Indicates whether or not we attempted to load this page previously
             _isRetryAttempt = _isRetryAttempt || false;
 
-            // Don't reload the page if it has already been loaded
-            if (page.$el) {
+            // Don't reload the page if all of its components have already been loaded
+            if (page.$el && page.$html && page.$text) {
                 return callback();
             }
+
+            // Variables to track progress of HTML and text content loads
+            var htmlDeferred = new $.Deferred();
```

Comment:
That makes sense. Lets go with the following pattern then:

```
var loadPage = function(page, callback, etc.) {
   // preload page
   // load html page
   loadPageHTML(page, function) {
    callback();
    loadPageText(page);
  }
}
```

That ensures that the HTML pages are loaded as fast as possible, and brings in the text version asynchronously. It also means that we don't really have to take the signature expiry scenario into account for the text loading function (as the html loading always just precedes it).